src/connectors/dynamic-widgets/connectDynamicWidgets.ts

```diff
@@ -36,10 +40,12 @@ export type DynamicWidgetsConnectorParams = {
    * Function to transform the items to render.
    * The function also exposes the full search response.
    */
-  transformItems?(
-    items: string[],
-    metadata: { results: SearchResults }
-  ): string[];
+  transformItems?: TransformItems<
+    string,
+    Omit<TransformItemsMetadata, 'results'> & {
+      results: NonNullable<TransformItemsMetadata['results']>;
+    }
```

Comment:
maybe we want to have an "override" internal type helper later, so it could be `type Override<TOriginal, TOverride> => Omit<TOriginal, keyof TOverride> & TOverride`, but not sure if we have the use case in other points already (possibly search for Omit in the code base, maybe index widget?)

Reply:
This is indeed a pattern used in a few files. I'll create a task in Jira with the current list I have so we can dedicate time for it.

Reply:
Done: [FX-780](https://algolia.atlassian.net/browse/FX-780).